generators/heroku/generator.mjs

```diff
@@ -87,491 +69,327 @@ export default class HerokuGenerator extends BaseGenerator {
+              throw new Error(stderr);
             }
-          } else {
-            done();
           }
-        });
-
-        child.stdout.on('data', data => {
-          const output = data.toString();
-          if (data.search('Heroku credentials') >= 0) {
-            this.abort = true;
-            this.log.error("Error: Not authenticated. Run 'heroku login' to login to your heroku account and try again.");
-            done();
-          } else {
-            this.log.verboseInfo(output.trim());
-          }
-        });
-      }),
-
-      herokuAddonsCreate: runAsync(function () {
-        if (this.abort) return;
-        const done = this.async();
-
-        const addonCreateCallback = (addon, err) => {
-          if (err) {
-            const verifyAccountUrl = 'https://heroku.com/verify';
-            if (_.includes(err, verifyAccountUrl)) {
-              this.abort = true;
-              this.log.error(`Account must be verified to use addons. Please go to: ${verifyAccountUrl}`);
-              this.log.error(err);
+        }
+      },
+
+      async herokuCreate() {
+        if (!this.hasHerokuCli || this.herokuAppExists) return;
+
+        const regionParams = this.herokuRegion !== 'us' ? ['--region', this.herokuRegion] : [];
+
+        this.log.log(chalk.bold('\nCreating Heroku application and setting up Node environment'));
+        const { stdout, stderr, exitCode } = await this.spawnHeroku(['create', this.herokuAppName, ...regionParams]);
+
+        if (stdout.includes('Heroku credentials')) {
+          throw new Error("Error: Not authenticated. Run 'heroku login' to log in to your Heroku account and try again.");
+        }
+
+        if (exitCode !== 0) {
+          if (stderr.includes('is already taken')) {
+            const prompts = [
+              {
+                type: 'list',
+                name: 'herokuForceName',
+                message: `The Heroku application "${chalk.cyan(this.herokuAppName)}" already exists! Use it anyways?`,
+                choices: [
+                  {
+                    value: 'Yes',
+                    name: 'Yes, I have access to it',
+                  },
+                  {
+                    value: 'No',
+                    name: 'No, generate a random name',
+                  },
+                ],
+                default: 0,
+              },
+            ];
+
+            this.log.log('');
+            const props = await this.prompt(prompts);
+            if (props.herokuForceName === 'Yes') {
+              await this.spawnHeroku(['git:remote', '--app', this.herokuAppName], { reject: true });
             } else {
-              this.log.verboseInfo(`No new ${addon} addon created`);
+              const { stdout } = await this.spawnHeroku(['create', ...regionParams]);
+              // Extract from "Created random-app-name-1234... done"
+              this.herokuAppName = stdout.substring(stdout.lastIndexOf('/') + 1, stdout.indexOf('.git'));
```

Comment:
If it’s not a git deployment?

Reply:
It still seems to work OK for this part.

```
? The Heroku application "play-more" already exists! Use it anyways? No, generate a random name
INFO! Creating app...
INFO!  done, evening-escarpment-92393
INFO!
INFO! https://evening-escarpment-92393-39f845122a90.herokuapp.com/ | https://git.heroku.com/evening-escarpment-92393.git
INFO!
INFO! set git remote heroku to https://git.heroku.com/evening-escarpment-92393.git
INFO!

Provisioning addons
```

However, there is a stacktrace further down:

```
Building application
ERROR! An error occured while running jhipster:heroku#productionBuild
ERROR! ERROR! Cannot read properties of null (reading 'on')
TypeError: Cannot read properties of null (reading 'on')
    at HerokuGenerator.printChildOutput (file:///Users/mraible/dev/generator-jhipster/dist/generators/heroku/generator.mjs:521:16)
    at HerokuGenerator.productionBuild (file:///Users/mraible/dev/generator-jhipster/dist/generators/heroku/generator.mjs:409:28)
    at HerokuGenerator.executeTask (file:///Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/dist/actions/lifecycle.js:241:26)
    at env.queueTask.once (file:///Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/dist/actions/lifecycle.js:215:56)
    at runLoop.add.once (file:///Users/mraible/dev/generator-jhipster/node_modules/yeoman-environment/dist/environment-base.js:381:23)
    at Immediate.<anonymous> (/Users/mraible/dev/generator-jhipster/node_modules/grouped-queue/lib/subqueue.js:48:34)
    at process.processImmediate (node:internal/timers:476:21)

> health-points@0.0.1-SNAPSHOT java:jar:prod
> npm run java:jar -- -Pprod
```

Reply:
The build finishes, but there's no deployment to Heroku that happens.